src/main/java/com/zaxxer/sansorm/internal/OrmReader.java

```diff
@@ -136,16 +136,14 @@ public static ResultSet statementToResultSet(final PreparedStatement stmt, final
    }
    // COMPLEXITY:ON
 
-   public static <T> T statementToObject(final PreparedStatement stmt, final Class<T> clazz, final Object... args) throws SQLException
+   public static <T> T statementToObject(final PreparedStatement stmt, final T target, final Object... args) throws SQLException
    {
       populateStatementParameters(stmt, args);
 
       try (final ResultSet resultSet = stmt.executeQuery()) {
          if (resultSet.next()) {
-            final T target = clazz.newInstance();
```

Comment:
Removal of this code would be a breaking change for us.

I would suggest leaving the original method signature, but modifying the implementation to:
```Java
public static <T> T statementToObject(final PreparedStatement stmt, final Class<T> clazz, final Object... args) throws SQLException {
   final T target = clazz.newInstance();
   return this. statementToObject(stmt, target, args);
}
```

Reply:
Sorry, you are right. I have missed that the method is public.